File: app/src/main/java/com/recursia/popularmovies/presentation/views/adapters/CastAdapter.java
package com.recursia.popularmovies.presentation.views.adapters;

import android.content.Context;
import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import com.bumptech.glide.Glide;
import com.bumptech.glide.load.resource.drawable.DrawableTransitionOptions;
import com.recursia.popularmovies.R;
import com.recursia.popularmovies.domain.models.Cast;
import com.recursia.popularmovies.utils.NetworkUtils;

import java.util.ArrayList;
import java.util.List;

import butterknife.BindView;
import butterknife.ButterKnife;

public class CastAdapter extends RecyclerView.Adapter<CastAdapter.CastViewHolder> {
    private static final int FADE_OUT_DURATION = 100; //ms

    private final Context context;
    private List<Cast> cast;

    public CastAdapter(Context context) {
        this.context = context;
        cast = new ArrayList<>();
    }

    public void setCast(List<Cast> cast) {
        this.cast = cast;
        notifyDataSetChanged();
    }

    @Override
    public int getItemCount() {
        return cast.size();
    }

    @NonNull
    @Override
    public CastViewHolder onCreateViewHolder(@NonNull ViewGroup viewGroup, int i) {
        View itemView = LayoutInflater.from(viewGroup.getContext()).inflate(R.layout.cast_item, viewGroup, false);
        return new CastViewHolder(itemView);
    }

    @Override
    public void onBindViewHolder(@NonNull final CastViewHolder movieViewHolder, int i) {
        Cast castItem = cast.get(i);
        //Image
        Glide.with(context)
                .load(NetworkUtils.getMediumProfileUrl(castItem.getProfilePath()))
                .placeholder(R.drawable.ic_user_placeholder)
                .transition(DrawableTransitionOptions.withCrossFade(FADE_OUT_DURATION))
                .into(movieViewHolder.castImage);
        //Name
        movieViewHolder.castName.setText(castItem.getName());
    }

    class CastViewHolder extends RecyclerView.ViewHolder {
        @BindView(R.id.castImage)
        ImageView castImage;
        @BindView(R.id.castName)
        TextView castName;

        CastViewHolder(@NonNull View itemView) {
            super(itemView);
            ButterKnife.bind(this, itemView);
        }
    }
}

File: app/src/main/java/com/recursia/popularmovies/presentation/views/adapters/CastAdapter.kt
package com.recursia.popularmovies.presentation.views.adapters

import android.content.Context
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import butterknife.BindView
import butterknife.ButterKnife
import com.bumptech.glide.Glide
import com.bumptech.glide.load.resource.drawable.DrawableTransitionOptions
import com.recursia.popularmovies.R
import com.recursia.popularmovies.domain.models.Cast
import com.recursia.popularmovies.utils.NetworkUtils
import java.util.*

class CastAdapter(private val context: Context) : RecyclerView.Adapter<CastAdapter.CastViewHolder>() {
    private var cast: List<Cast> = ArrayList()

    fun setCast(cast: List<Cast>) {
        this.cast = cast
        notifyDataSetChanged()
    }

    override fun getItemCount(): Int {
        return cast.size
    }

    override fun onCreateViewHolder(viewGroup: ViewGroup, i: Int): CastViewHolder {
        val itemView = LayoutInflater.from(viewGroup.context).inflate(R.layout.cast_item, viewGroup, false)
        return CastViewHolder(itemView)
    }

    override fun onBindViewHolder(movieViewHolder: CastViewHolder, i: Int) {
        val castItem = cast[i]
        // Image
        Glide.with(context)
                .load(NetworkUtils.getMediumProfileUrl(castItem.profilePath ?: ""))
                .placeholder(R.drawable.ic_cast_placeholder)
                .transition(DrawableTransitionOptions.withCrossFade(FADE_OUT_DURATION))
                .into(movieViewHolder.castImage)
        // Name
        movieViewHolder.castName.text = castItem.name
    }

    inner class CastViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        @BindView(R.id.castImage)
        lateinit var castImage: ImageView
        @BindView(R.id.castName)
        lateinit var castName: TextView

        init {
            ButterKnife.bind(this, itemView)
        }
    }

    companion object {
        private const val FADE_OUT_DURATION = 100 // ms
    }
}
